tuxemon/states/combat/combat.py

```diff
@@ -870,13 +874,22 @@ def trigger_player_run(self, player):
     def end_combat(self):
         """End the combat"""
         # TODO: End combat differently depending on winning or losing
+        old_stats_data = Technique.keep_old_stats(self)
```

Comment:
this code is hard to follow, and made worse because the old stats values are spread around the techniques and now the combat class.  it would be more clear if the monsters were checked for statuses, and if the status has changed their stats, then revert the stats using the old stats value from the technique.

Reply:
if you guys have a more efficient way of doing it thats fine, i admit this code turned out a bit hard to follow as i kept having issues and making changes. it does work though

Reply:
this is going to be a source of problems in the future.  please add a comment so that future developers can narrow down issues in battles with multiple trainers and strange stats changes.
```
# The values here are set in the technique and do not contain reference to trainers.  This code needs to be updated to support multiple trainers.
```